rotkehlchen/chain/ethereum/adex/utils.py

```diff
@@ -0,0 +1,130 @@
+)
+STAKING_ADDR = '0x4846C6837ec670Bbd1f5b485471c8f64ECB9c534'
+CREATE2_SALT = f'0x{bytearray(32).hex()}'
+ADX_AMOUNT_MANTISSA = 10**18
+
+# Pools data
+TOM_POOL_ID = HexStr('0x2ce0c96383fb229d9776f33846e983a956a7d95844fac57b180ed0071d93bb28')
+POOL_ID_POOL_NAME = {
+    TOM_POOL_ID: 'Tom',
+}
+
+# Tom pool fee rewards API constants
+TOM_POOL_FEE_REWARDS_API_URL = 'https://tom.adex.network/fee-rewards'
+SECONDS_PER_YEAR = FVal('31536000')
+PERIOD_END_AT_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+ADEX_EVENTS_PREFIX = 'adex_events'
+
+
+def deserialize_adex_event_from_db(
+        event_tuple: AdexEventDBTuple,
+) -> Union[Bond, Unbond, UnbondRequest]:
+    """Turns a tuple read from DB into an appropriate AdEx event.
+    May raise a DeserializationError if something is wrong with the DB data.
+
+    Event_tuple index - Schema columns
+    ----------------------------------
+    0 - tx_hash
+    1 - address
+    2 - identity_address
+    3 - timestamp
+    4 - bond_id
+    5 - type
+    6 - pool_id
+    7 - amount
+    8 - nonce
+    9 - slashed_at
+    10 - unlock_at
+    """
+    db_event_type = event_tuple[5]
+    if db_event_type not in {str(event_type) for event_type in EventType}:
+        raise DeserializationError(
+            f'Failed to deserialize event type. Unknown event: {db_event_type}.',
+        )
+
+    tx_hash = HexStr(event_tuple[0])
+    address = deserialize_ethereum_address(event_tuple[1])
+    identity_address = deserialize_ethereum_address(event_tuple[2])
+    timestamp = deserialize_timestamp(event_tuple[3])
+    bond_id = HexStr(event_tuple[4])
+
+    if db_event_type == str(EventType.BOND):
+        try:
```

Comment:
Hmm that is very wrong. Assertion should only be used for programming errors. Keep in mind assertsion can be turned off in optimized code!!!!

And putting assertions, then catching the assertion error and raising a DeserializationError is unnecessarily complicated.

Just do (pseudocode):

```python
if any(event_tuple[x] is None for x in (6, 7, 8, 9)):
    raise DeserializationError(
                f'Failed to deserialize bond event. Unexpected data: {event_tuple}.',
            ) 
```